apps/dataset/views/document.py

```diff
@@ -238,6 +238,24 @@ def put(self, request: Request, dataset_id: str, document_id: str):
                     request.data
                 ))
 
+        class Batch(APIView):
+            authentication_classes = [TokenAuth]
+
+            @action(methods=['PUT'], detail=False)
+            @swagger_auto_schema(operation_summary="批量取消任务",
+                                 operation_id="批量取消任务",
+                                 request_body=DocumentApi.BatchCancel.get_request_body_api(),
+                                 manual_parameters=DocumentSerializers.Create.get_request_params_api(),
+                                 responses=result.get_default_response(),
+                                 tags=["知识库/文档"]
+                                 )
+            @has_permissions(
+                lambda r, k: Permission(group=Group.DATASET, operate=Operate.MANAGE,
+                                        dynamic_tag=k.get('dataset_id')))
+            def put(self, request: Request, dataset_id: str):
+                return result.success(
+                    DocumentSerializers.Batch(data={'dataset_id': dataset_id}).batch_cancel(request.data))
+
     class Refresh(APIView):
         authentication_classes = [TokenAuth]
 
```

Comment:
There appears to be a few areas that need attention:

1. **Class Declaration**: The `Batch` class declaration does not seem to have been properly indented relative to other classes.

2. **Method Definition**: The `put` method definition within the `Batch` class is missing an opening curly brace (`{`).

3. **Response Handling**: There seems to be an issue with how the response from `result.success()` is being defined or returned.

Here's a corrected version of your code:

```python
@@ -238,6 +238,25 @@ def put(self, request: Request, dataset_id: str, document_id: str):
                     request.data
                 ))

 class Batch(APIView):
     authentication_classes = [TokenAuth]
 
-    @action(methods=['PUT'], detail=False)
+    @action(methods=['PUT'], detail=False)
+    @swagger_auto_schema(operation_summary="批量取消任务",
+                         operation_id="批量取消任务",
+                         request_body=DocumentApi.BatchCancel.get_request_body_api(),
+                         manual_parameters=DocumentSerializers.Create.get_request_params_api(),
+                         responses=result.get_default_response(),
+                         tags=["知识库/文档"]
+                         )
+    @has_permissions(
+        lambda r, k: Permission(group=Group.DATASET, operate=Operate.MANAGE,
+                                   dynamic_tag=k.get('dataset_id')))
+    def put(self, request: Request, dataset_id: str) -> JsonResponse:
+        # Ensure you are returning a JsonResponse object here
+        return result.success(
+            DocumentSerializers.Batch(data={'dataset_id': dataset_id}).batch_cancel(request.data))
 

 class Refresh(APIView):
     authentication_classes = [TokenAuth]

```

### Key Corrections:
- **Indentation**: Corrected the indentation of the `Batch` class and its methods.
- **Return Type**: Ensured that the `return` statement returns a `JsonResponse`, which is expected by Django Rest Framework when using `@swagger_auto_schema`.

These changes should resolve any syntax errors or misconfigurations in your code.